File: php/src/Behavioral/Iterator/DvdListIterator.php
<?
  
require_once 'IteratorInf.php';

 
class DvdListIterator implements IteratorInf
{
	private $currentPosition;
	private $list;

	function __construct($list)
	{
		$this->list = $list;
		$this->currentPosition = 0;
	}
			 
	function first()
	{
		$this->currentPosition = 0;
	}
			 
	function next()
	{
		if($this->currentPosition < ($this->list->getTitleCount()-1))
			$this->currentPosition++;
	}
			 
	function isDone ()
	{
		return $this->currentPosition >= ($this->list->getTitleCount()-1);
	}
			 
	function currentItem()
	{
		return $this->list->getList()[ $this->currentPosition ];
	}


}

File: python/src/Behavioral/Iterator/DvdListIterator.py


#//DvdListIterator
from IteratorInterface import IteratorInterface 

class DvdListIterator(IteratorInterface):

	def __init__(self,list=None):
		self.currentPosition = 0
		if list is None :
			self.list = []
		else :
			self.list = list
			
	def first(self) : 
		self.currentPosition = 0
			 
	def next(self) :
		if self.currentPosition < self.list.titleCount:
			self.currentPosition += 1 
			 
	def isDone(self) : 
		return self.currentPosition >= self.list.titleCount
			 
	def currentItem(self) : 
		return self.list.titles[ self.currentPosition ]
